web/app/js/components/DeploymentDetail.jsx

```diff
@@ -249,10 +241,12 @@ export default class DeploymentDetail extends React.Component {
         { !this.state.error ? null : <ErrorBanner message={this.state.error} /> }
         { !this.state.loaded ? <ConduitSpinner /> :
           <div>
-            <div className="page-header">
-              <div className="subsection-header">Deployment detail</div>
-              {this.renderDeploymentTitle()}
-            </div>
+            <PageHeader
+              subHeaderTitle="Deployment detail"
+              subHeader={this.renderDeploymentTitle()}
+              subMessage={incompleteMeshMessage(this.state.deploy)}
```

Comment:
I think this needs to be:

```
subMessage={!this.state.added ? incompleteMeshMessage(this.state.deploy) : null}
```

`incompleteMeshMessage` never returns null, but the render method in the PageHeader component has:

```
{!this.props.subMessage ? null : <div>{this.props.subMessage}</div>}
```

That results in us displaying the incomplete mesh message on every deployment page, even for those that have been added to the mesh. For instance:

![image](https://user-images.githubusercontent.com/9226/35759029-0202dc7e-082c-11e8-85ae-0123dc466162.png)

Reply:
whoops, yes, that check got lost in the porting. nice catch!